ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -149,6 +160,65 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req abci.Reque
 	return nil
 }
 
+// adjustAmountBySpreadFactor adjusts the amount by the spread factor.
+// This is done to adjust the amount of tokens in the token_swapped event by the spread factor,
+// as the amount in the event is the amount AFTER the spread factor is applied.
+// therefore, we need to adjust the amount by the spread factor to get the amount BEFORE the spread factor is applied.
+// NOTE: This applies to CL pools only
+func (s *indexerStreamingService) adjustTokenInAmountBySpreadFactor(ctx context.Context, event *abci.Event) error {
+	if event.Type != "token_swapped" {
+		return nil
+	}
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	var poolIdStr string
+	var afterTokensIn string
+	var afterTokensInIndex int
+	attributes := event.Attributes
+	// Find the pool id and tokens in attributes from the token_swapped
+	for index, attribute := range attributes {
+		if poolIdStr != "" && afterTokensIn != "" {
+			break
+		}
+		if attribute.Key == "pool_id" {
+			poolIdStr = attribute.Value
+		}
+		if attribute.Key == "tokens_in" {
+			afterTokensIn = attribute.Value
+			afterTokensInIndex = index
+		}
+	}
+	if poolIdStr == "" || afterTokensIn == "" {
+		return errors.New("pool id or tokens in not found")
+	}
+	poolId, err := strconv.ParseInt(poolIdStr, 10, 64)
+	if err != nil {
+		return errors.New("failed to parse pool id")
+	}
+	poolType, err := s.keepers.PoolManagerKeeper.GetPoolType(sdkCtx, uint64(poolId))
+	if err != nil {
+		return errors.New("failed to get pool type")
+	}
+	// Adjustment required only for CL pools
+	if poolType != pooltypes.Concentrated {
+		return nil
+	}
+	// Get the pool and its spread factor
+	pool, err := s.keepers.PoolManagerKeeper.GetPool(sdkCtx, uint64(poolId))
+	if err != nil {
+		return errors.New("failed to get pool")
+	}
+	spreadFactor := pool.GetSpreadFactor(sdkCtx)
+	coins, err := sdk.ParseCoinsNormalized(afterTokensIn)
+	if err != nil {
+		return errors.New("failed to parse tokens in")
+	}
+	tokenInAmount := coins[0].Amount.ToLegacyDec()
+	// Adjust the amount by the spread factor, i.e. before = after/(1 - spreadFactor)
+	adjustedAmt := tokenInAmount.Quo(math.LegacyNewDec(1).Sub(spreadFactor))
```

Comment:
Let's define `osmomath.OneDec()` in the global space to avoid re-initializing (which is expensive)